apps/sv/frontend/src/components/forms/CreateProposalform.tsx

```diff
@@ -4,55 +4,112 @@
-import { useState } from 'react';
+import { dateTimeFormatISO, JSONValue } from '@lfdecentralizedtrust/splice-common-frontend-utils';
+import { useMemo, useState } from 'react';
 import { createProposalActions } from '../../utils/governance';
+import { useDsoInfos } from '../../contexts/SvContext';
+import { DsoRulesConfig } from '@daml.js/splice-dso-governance/lib/Splice/DsoRules';
+import { buildDsoConfigChanges } from '../../utils/buildDsoConfigChanges';
+import { useAppForm } from '../../hooks/form';
+
+type ConfigFormData = Record<string, string>;
 
 type CreateProposalAction = (typeof createProposalActions)[number];
 
 interface CreateProposalFormProps {
   action: CreateProposalAction;
 }
 
+type FormValues = {
+  action: string;
+  expiryDate: string;
+  effectiveDate: string;
+  url: string;
+  summary: string;
+  member: string;
+} & ConfigFormData;
+
+const tomorrow = () => dayjs().add(1, 'day').format(dateTimeFormatISO);
+
 export const CreateProposalForm: React.FC<CreateProposalFormProps> = ({ action }) => {
   const [searchParams, _] = useSearchParams();
   const actionFromParams = searchParams.get('action');
   const [effectivityType, setEffectivityType] = useState('custom');
+  const dsoInfoQuery = useDsoInfos();
 
   const memberOptions: { key: string; value: string }[] = [
     { key: 'sv1', value: 'Super Validator 1' },
     { key: 'sv2', value: 'Super Validator 2' },
   ];
 
-  const form = useForm({
-    defaultValues: {
+  const defaultValues = useMemo((): FormValues => {
+    if (!dsoInfoQuery.data) {
+      return {
+        action: action.name,
+        expiryDate: dayjs().format(dateTimeFormatISO),
+        effectiveDate: tomorrow(),
+        url: '',
+        summary: '',
+        member: '',
+      };
+    }
+
+    const dsoConfig = dsoInfoQuery.data.dsoRules.payload.config;
+    const dsoCurrentConfig = DsoRulesConfig.encode(dsoConfig) as Record<string, JSONValue>;
+    const dsoConfigChanges = buildDsoConfigChanges(dsoConfig, dsoConfig, true);
+    console.log('dsoConfig', dsoConfig);
+    console.log('dsoCurrentConfig', dsoCurrentConfig);
+    console.log('dsoConfigChanges', dsoConfigChanges);
+
+    return {
       action: action.name,
-      expiryDate: dayjs(),
-      effectiveDate: dayjs().add(1, 'day'),
+      expiryDate: dayjs().format(dateTimeFormatISO),
+      effectiveDate: tomorrow(),
       url: '',
       summary: '',
       member: '',
+      ...dsoConfigChanges.reduce((acc, field) => {
+        acc[field.fieldName] = field.currentValue;
+        return acc;
+      }, {} as ConfigFormData),
+    };
+  }, [dsoInfoQuery.data, action.name]);
+
+  const form = useAppForm({
+    defaultValues,
+    onSubmit: async ({ value }) => {
+      const data = { ...value, effectiveDate: normalizeEffectivity(value.effectiveDate) };
+      console.log('form submit', data);
     },
   });
 
+  const normalizeEffectivity = (date: string) => {
```

Comment:
The backend considers an undefined effective date as effective at threshold. This function helps with that